node/src/reactor/participating/error.rs

```diff
@@ -39,4 +44,50 @@ pub enum Error {
     /// Failed to serialize data.
     #[error("serialization: {0}")]
     Serialization(#[source] bincode::ErrorKind),
+
+    /// Engine state error.
+    #[error("engine state: {0}")]
```

Comment:
We could use `transparent` for this and the next two.

```suggestion
    #[error(transparent)]
```